### PR TITLE
feature: Background updates on mobile

### PR DESCRIPTION
**Pull request by mananjadhav**
cc - @Julesssss 

### Fixed Issues

$ https://github.com/Expensify/App/issues/4264

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

**Comment by Julesssss** [comment]
> Sure. Quick question outside the PR, do we have a process where if contributors want to go on vacation they can notify?

We don't at the moment, but I don't think it's a problem here as I've caused some delays here myself.

**Comment by Julesssss** [comment]
Hey @mananjadhav. Once again, sorry for my long delay here. Now that I'm back and caught up I will make sure not to delay us again. 

Do you remember what exactly was failing? I'm seeing a lot of these errors in both iOS and Android:

Screenshot 2021-12-09 at 11 40 19

**Comment by mananjadhav** [comment]
I'll check and get back to you. But as far as I remember the task was getting initialized but was getting timed out.

File: index.js [hunk 1]
```diff
@@ -4,9 +4,42 @@
...
+import BackgroundFetch from 'react-native-background-fetch';
+
 import App from './src/App';
 import Config from './src/CONFIG';
 import additionalAppSetup from './src/setup';
 
+
+const headlessTask = async (event) => { // eslint-disable-line
+
+    // Get task id from event {}:
```
Comment (kidroca) [inline index.js:16]: The task should not be defined here in `index.js`
It should be defined in `libs` and then registered during `setup` or in `Expensify.js` `didMount` phase

---

I see this is already done, and there's a lib about this, so the code here seems like a leftover that should be removed

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -282,9 +286,9 @@
 						TestTargetID = 13B07F861A680F5B00A75B9A;
 					};
 					13B07F861A680F5B00A75B9A = {
-						DevelopmentTeam = 368M544MTT;
+						DevelopmentTeam = 72TRNPHV7D;
 						LastSwiftMigration = 1230;
-						ProvisioningStyle = Manual;
+						ProvisioningStyle = Automatic;
```
Comment (kidroca) [inline ios/NewExpensify.xcodeproj/project.pbxproj:291]: This should probably be reverted

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 3]
```diff
@@ -841,10 +847,10 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = NewExpensify/Chat.entitlements;
-				CODE_SIGN_IDENTITY = "iPhone Distribution";
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "Apple Development";
+				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 3;
-				DEVELOPMENT_TEAM = 368M544MTT;
+				DEVELOPMENT_TEAM = 72TRNPHV7D;
```
Comment (kidroca) [inline ios/NewExpensify.xcodeproj/project.pbxproj:853]: This should definitely be reverted

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -858,7 +864,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.chat.expensify.chat;
 				PRODUCT_NAME = "New Expensify";
-				PROVISIONING_PROFILE_SPECIFIER = chat_expensify_appstore;
+				PROVISIONING_PROFILE_SPECIFIER = "";
```
Comment (kidroca) [inline ios/NewExpensify.xcodeproj/project.pbxproj:867]: This should definitely be reverted

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 5]
```diff
@@ -874,10 +880,10 @@
 				ASSETCATALOG_COMPILER_APPICON_NAME = AppIcon;
 				CLANG_ENABLE_MODULES = YES;
 				CODE_SIGN_ENTITLEMENTS = NewExpensify/Chat.entitlements;
-				CODE_SIGN_IDENTITY = "iPhone Distribution";
-				CODE_SIGN_STYLE = Manual;
+				CODE_SIGN_IDENTITY = "Apple Development";
+				CODE_SIGN_STYLE = Automatic;
 				CURRENT_PROJECT_VERSION = 3;
-				DEVELOPMENT_TEAM = 368M544MTT;
+				DEVELOPMENT_TEAM = 72TRNPHV7D;
```
Comment (kidroca) [inline ios/NewExpensify.xcodeproj/project.pbxproj:886]: This should definitely be reverted

File: ios/NewExpensify.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -890,7 +896,7 @@
 				);
 				PRODUCT_BUNDLE_IDENTIFIER = com.chat.expensify.chat;
 				PRODUCT_NAME = "New Expensify";
-				PROVISIONING_PROFILE_SPECIFIER = chat_expensify_appstore;
+				PROVISIONING_PROFILE_SPECIFIER = "";
```
Comment (kidroca) [inline ios/NewExpensify.xcodeproj/project.pbxproj:899]: This should definitely be reverted

**Review by kidroca — CHANGES_REQUESTED** [review]
Left some notes mostly on changes that shouldn't have been committed

**Comment by botify** [comment]
@robertjchen looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

**Comment by robertjchen** [comment]
Wait, what happened here? I merged https://github.com/Expensify/App/pull/6641 but somehow this was merged as well? 🤔

**Comment by robertjchen** [comment]
I'm going to revert this- this was not supposed to be merged.

**Comment by Julesssss** [comment]
Wow, that was unexpected! 

It's my fault, as [my PR](https://github.com/Expensify/App/pull/5953) accidentally branched from this PR. I made a lot of changes and rebased a couple of times, which made removing these commits very time-consuming (I didn't notice this immediately).

I'll get this PR reverted

**Comment by mananjadhav** [comment]
Wooahh, let me close this PR and raise one more as draft that shouldn't have allow merging?

**Comment by Julesssss** [comment]
Hi @mananjadhav. Don't worry, I think you can just create a new PR. You might need to cherry-pick the commits though, so they have a new commit ID.

I had a large PR that accidentally branched from yours, I tried to rebase them away but was unsuccessful as it was such a large PR with many conflicts over the last month.  

I think the draft status was ignored by Github as technically your commits are now in `main` -- but they are not aware that I reverted in my later commits

**Comment by mananjadhav** [comment]
I'll cherry-pick or move the commits as a fresh set of changes. Tracking my branch separately.

**Comment by Julesssss** [comment]
Thanks, @mananjadhav. Sorry for the confusion.

**Pull request merged**